### ISSUE TITLE
Missing Logs from Containers in a CrashLoop

### ISSUE THREAD

**Issue by chen-anders** [zalando-incubator/kubernetes-log-watcher#29]
I've noticed that logs from containers that may have been in a crash-loop do not get collected. The observed behavior is that a new "pause" docker container (whose logs get ignored) gets created under a different container ID (in docker) when the crashloop is detected and that somehow the logs of the prior crashing container are abandoned.

Test crash-looping container:

```
apiVersion: v1
kind: Pod
metadata:
  name: crashloop
  labels:
    application: crashloop
    version: '1.0'
spec:
  containers:
  - name: container
    image: python:3.5
    command: ["/bin/sh"]
    args: ["-c", "sleep 3; echo 'hello'; exit 1;"]
```

We currently have our logs going up to Scalyr.

**Comment by chen-anders** [comment]
An idea kinda popped up: 


Excerpt of `kube.py`: 
```
kwargs['log_file_name'] = os.path.basename(container['log_file'])
kwargs['log_file_path'] = container['log_file']
```

k8s stores logs for a longer period of time in `/var/log/containers`. If we can somehow get the correct log paths and leverage this different directory instead - this may work in our luck.

**Comment by mohabusama** [comment]
Hi @chen-anders 

Thanks for the report. To fix this issue we need:

1. Scalyr agent ability to ship logs from the beginning of the log file https://github.com/scalyr/scalyr-agent-2/issues/48
2. Kubernetes-log-watcher should keep the container logs for a while after the container is terminated

Will keep this thread updated with the progress.

**Comment by chen-anders** [comment]
This is solved now with #32.
